spec/requests/event_spec.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Event endpoint', type: :request do
+  let!(:admin) { create(:admin) }
+  let!(:user) { create(:user) }
+  let!(:event) { create(:event) }
+  let!(:event_to_update) { create(:event) }
+  let!(:auth_headers) { admin.create_new_auth_token }
+  let!(:auth_headers_user) { user.create_new_auth_token }
+  let!(:invitation) { create(:invitation, user: user, event: event) }
+
+  let!(:event_to_create_with_user) do
+    {
+      'event' =>
+        {
+          'name' => 'evento_testing',
+          'address' => 'testing_address333',
+          'cost' => 200,
+          'start_time' => '2020-10-07T13:28:06.419Z',
+          'end_time' => '2020-10-07T15:28:06.419Z',
+          'cancelled' => false
+        },
+      'user_ids' => [user.id.to_i]
+    }
+  end
+  let!(:event_to_create_without_user) do
+    {
+      'event' =>
+        {
+          'name' => 'evento_testing',
+          'address' => 'testing_address',
+          'cost' => 200,
+          'start_time' => '2020-10-07T13:28:06.419Z',
+          'end_time' => '2020-10-07T15:28:06.419Z',
+          'cancelled' => false
+        }
+    }
+  end
+  let!(:event_data_update) do
+    {
+      'event' =>
+        {
+          'name' => 'evento_update',
+          'address' => 'evento_update',
+          'cost' => 500,
+          'start_time' => '2020-10-07T13:28:06.419Z',
+          'end_time' => '2020-10-07T15:28:06.419Z',
+          'cancelled' => false
+        },
+      'user_ids' => [user.id]
+    }
+  end
+  describe 'GET /api/v1/admin/events' do
+    context 'return all events' do
+      it 'return 200 ok' do
+        get api_v1_admin_events_path, headers: auth_headers
+        expect(response).to have_http_status(200)
+        events_response = Oj.load(response.body)['events']
+        expect(events_response[0].except('users')).to include(event.as_json.except('updated_at', 'created_at'))
+        expect(events_response[0]['users'][0]['id']).to eq(invitation.as_json['user_id'])
```

Comment:
It would be really good to try this endpoint with at least two events (as well as with none)